### PR TITLE
qt: Generalized css files, simple design changes, added scripts to keep track of color usage

### PR DESCRIPTION
**Pull request by xdustinface**
Prior to this PR `dark.css` and `light.css` contained a lot shared layout/style changes which were required to kept synced in both files when a design change happened. This PR generalises those shared layout/style changes and puts them into `general.css` which is loaded before the actual theme css file. 

~~This is currently based on #3500 so the actual first commit of this PR is 5530541. Will rebase on `develop` as soon as the other PR is merged.~~

#### Generalizations and some redesigns 
e47ef8c to 2d5610b are the ones putting all shared changes into `general.css`. They also contain some simple redesigns for that see the individual commit or below the before/after images. 

#### Cleanup
06482d9 to cadfd7f are cleaning up the css files, giving them the same structure and fix color issues happened due to the generalizations. 

- First part of the css file contains general Qt standard class changes sorted by the alphabet.
- Second part of the file contains custom classes also sorted by the alphabet.

e64e046 adds copyright and a file description to all css files 

#### Script

84cbfad adds the python script `update_colors.py` which parses all the css files, generates a list of used colors and puts them into the appropriate css file's description section between `<colors></colors>` tags. It also generates the files in `src/qt/res/css/colors/`. They contain some expandable "analyses" of the color usage in the css files. Currently those files contain:

- List of used colors
- List of attribute/color combination with a list of selectors they are used by. 
- List of colors with a list of attributes they are used by. 

0d680abfab4649c2539f37d5ebe7c8fc77cc9ed2 just adds the results of running `update_colors.py` 

## Before/After

Here are some before/after images with the most significant changes highlighted in the light theme (they apply the same to the dark) compared to #3500.

### Light

ModalOverlay Before             |  ModalOverlay Now
:-------------------------:|:-------------------------:
 | 

Overview Before             |  Overview Now
:-------------------------:|:-------------------------:
 | 

Send Before             |  Send Now
:-------------------------:|:-------------------------:
 | 

Receive Before             |  Receive Now
:-------------------------:|:-------------------------:
 | 

CoinControl Before             |  CoinControl Now
:-------------------------:|:-------------------------:
 | 

### Dark

ModalOverlay Before             |  ModalOverlay Now
:-------------------------:|:-------------------------:
 | 

Overview Before             |  Overview Now
:-------------------------:|:-------------------------:
 | 

Send Before             |  Send Now
:-------------------------:|:-------------------------:
 | 

Receive Before             |  Receive Now
:-------------------------:|:-------------------------:
 | 

CoinControl Before             |  CoinControl Now
:-------------------------:|:-------------------------:
 |

### CODE REVIEW

**Comment by UdjinM6** [comment]
Just merged #3500 , needs rebase now.

**Comment by xdustinface** [comment]
Done! Also updated the initial description.

**Comment by xdustinface** [comment]
Good points 👍 Changed, and yeah you got it right, those files are just helpers 🙂 Removed them from the repo and added them to `.gitignore`

File: src/qt/res/css/trad.css [hunk 1]
```diff
@@ -1,29 +1,68 @@
...
+*/
+
+/******************************************************
+ModalOverlay
+******************************************************/
+
+QWidget#bgWidget { /* The frame overlaying the overview-page */
+    padding-left: 10px;
+    padding-right: 10px;
+    background-color: #fafafacc;
```
Comment (UdjinM6) [inline src/qt/res/css/trad.css:35]: This results in some weird yellow-ish background. I guess you messed up with the alpha channel here, probably want smth like
```suggestion
    background-color: #ccfafafa;
```
instead (a slightly transparent white-ish one).

Comment (xdustinface) [reply src/qt/res/css/trad.css:35]: Oh right, good one, had that issue in the other files too.. seems like it forgot to fix it in the trad.css 👍

**Comment by xdustinface** [comment]
Thanks for the updates to the script, picked them all 👍

**Comment by UdjinM6** [comment]
Looks mostly good. One issue I noticed is that there are two red colors in UI now - `red` and `#ff4500`. The simple `red` one is too bright for the Dark theme and the lighter `#ff4500` looks ok in all themes imo. Let's use the later one only maybe? See 400058e686

**Comment by xdustinface** [comment]
I have another pull requesting coming up soon which fixes your issue with the red and brings overall more consistency to layouts and coloring. Is it good enough to see this as accepted issue in this PR? 😄 I will be ready with the other PR probably today or tomorrow!

**Comment by PastaPastaPasta** [comment]
This is a massive PR, imo it probably makes sense to split it out into smaller PRs or in future smaller PRs to make review easier

**Comment by xdustinface** [comment]
Isn't it splitting the PR the same (at least for reviewing) as if you just make use of the commit selection feature? Tell me for the future, how would you have split this to make it more appropriate?

To give you my reason why i chose to push this all into this PR: Just to have an as much as possible clean transition for from "layout changes in `dark.css` **and** `light.css` **and** `general.css`" to "layout changes only in `general.css`"  and to have all css files in the same order in one commit later on `develop`

**Comment by PastaPastaPasta** [comment]
It's hard for me to get the motivation to review 33 commits, so splitting it up makes it a bit easier emotionally :joy: 

To me most of the commits seem atomic, so you probably could have split it a number of ways. The update colors script and associated stuff probably could've been in a separate PR.

**Comment by xdustinface** [comment]
haha there were only 23 commits when i submitted! 😅 And wouldn't you still have all 33 pop up in the last PR because they all are based on each other if i would split this up now? 🤔

Please don't get me wrong, in general i totally get the idea of small pull requests though in this case i somehow just felt like i need to push this all into one "cleanup-commit" which contains all css restructuring i had in mind, sorry 🙈 

Anyway.. if you still reeaallly want me to split this one, let me know i will do it for sure!

**Comment by PastaPastaPasta** [comment]
Nah, this PR is probably fine. I gave it a code review, will test tomorrow or next day

**Comment by PastaPastaPasta** [comment]
Some things I noticed, 
the back/white doesn't look as clean
(see below, second comment)



It starts out smaller
starts smaller
text on send/receive pages are too large
text too large

"automatically selected" on send page is black instead of grey
automaticallyselected

selected amount text looks worse

text large

**Comment by PastaPastaPasta** [comment]
Had to use phone to take picture because for whatever reason screenshot shows it as basically the same color, but to my eye/on my monitor it's clearly not

IMG_20200612_181328

**Comment by xdustinface** [comment]
> text on send/receive pages are too large

don't you like the increase? That was intentional because i rather thought they were too small before at least without those boxes around it.. 😅 

I have changes to all this already coming in other PRs very soon. Im currently still testing and fixing for windows/linux will push all that once its sorted. IMO it would make sense to leave this one here as it is if there are no more changes to the python script required.

**Comment by PastaPastaPasta** [comment]
I think they look too big now... Idk... Maybe it's fine

What about the other issues?

**Pull request merged**